### ISSUE TITLE
C++: Support the array syntax in the code generator

### ISSUE THREAD

**Issue by cmnrd** [icyphy/lingua-franca#136]

**Comment by cmnrd** [comment]
After looking into this for a couple of days now and giving it a good thought, I decided to not support the array syntax in C++. Plain C style arrays are difficult to handle and there is no sensible way of 
generating code for them. After trying to circumvent some of the problems, I reached the conclusion that it would be best to drop support for arrays altogether. As a replacement for C style arrays, the containers `std::vector` or `std::array` should be used. This produces cleaner and safer code.

I was also considering to automatically translate `type[]` to `std::vector<type>` and `type[42]` to `std::array<type, 42>`. However, I think this could lead to some unexpected results as the array syntax of LF is easy to confuse with the array syntax of C/C++. I think it is better to be explicit here and force the programmer to declare what she whants instead of guessing it for her.

I still plan adapt the array based tests from C to C++ as they will show how similar functionality can be achieved using C++ containers.

**Comment by cmnrd** [comment]
After a discussion with Marten I reconsidered the above statement and decided to map fixed size lists (e.g. `int[4]`) to C++ fixed size arrays (e.g. `std::array<int, 3>`) and variable sized lists (e.g. `int[]`) to `std::vector<int>`. The motivation for this reconsideration, is mostly the special meaning of `time` in LF. It is definetly clearer to write `time[]` then beeing forced to write `std::vector<reactor::Duration>` which requires some deaper knowledge of how the C++ framework which is used in the background. 

Support for fixed and variable size lists in C++ was added in  #144. Mapping the LF list notation to C++ containers provides clear semantics avoiding the pitfalls of plain C-style arrays.
